ui/pages/confirmations/hooks/alerts/useConfirmationOriginAlerts.ts

```diff
@@ -0,0 +1,49 @@
+import { useMemo } from 'react';
+import { useSelector } from 'react-redux';
+import { TransactionMeta } from '@metamask/transaction-controller';
+
+import { Alert } from '../../../../ducks/confirm-alerts/confirm-alerts';
+import { Severity } from '../../../../helpers/constants/design-system';
+import { useI18nContext } from '../../../../hooks/useI18nContext';
+import { Confirmation, SignatureRequestType } from '../../types/confirm';
+import { currentConfirmationSelector } from '../../selectors';
+import { isValidASCIIURL, toPunycodeURL } from '../../utils/confirm';
+import { isSignatureTransactionType } from '../../utils';
+
+const useConfirmationOriginAlerts = (): Alert[] => {
+  const t = useI18nContext();
+
+  const currentConfirmation: Confirmation | undefined = useSelector(
+    currentConfirmationSelector,
+  );
+
+  return useMemo<Alert[]>((): Alert[] => {
+    if (!currentConfirmation) {
+      return [];
+    }
+
+    const origin = isSignatureTransactionType(currentConfirmation)
+      ? (currentConfirmation as SignatureRequestType).msgParams?.origin
+      : (currentConfirmation as TransactionMeta).origin;
+
+    if (origin && isValidASCIIURL(origin)) {
+      return [];
+    }
+
+    return [
+      {
+        key: 'originSpecialCharacterWarning',
+        reason: t('addressMismatch'),
+        field: 'originSpecialCharacterWarning',
```

Comment:
As mentioned in the other comments, could we instead just have a dynamic `field` here based on the confirmation type since we can't have both simultaneously?

Then we can use actual field references such as `requestFrom` or `origin` via `RowAlertKey.

Reply:
Fixed in the PR